server/service/osquery_utils/queries.go

```diff
@@ -705,98 +705,111 @@ var scheduledQueryStats = DetailQuery{
 	DirectTaskIngestFunc: directIngestScheduledQueryStats,
 }
 
+var softwareLinuxDebian = DetailQuery{
+	Query: `
+		SELECT
+		name AS name,
+		version AS version,
+		'Package (deb)' AS type,
+		'deb_packages' AS source,
+		'' AS release,
+		'' AS vendor,
+		'' AS arch,
+		'' AS installed_path
+		FROM deb_packages
+		WHERE status = 'install ok installed'`,
+	Platforms:        fleet.LinuxOS["debian"],
+	DirectIngestFunc: directIngestSoftware,
```

Comment:
AFAICS, splitting software query will cause the following:
`directIngestSoftware` calls `ds.UpdateHostSoftware` which will remove everything that's not `deb_packages` for this host (`ds.UpdateHostSofware` calls `deleteUninstalledHostSoftwareDB`).

If we want to split software queries into multiple queries (which we plan to do in the near future for performance reasons - query denylisting in particular) we need to call `directIngestSoftware` with all the results from all the now splitted software queries; or refactor our ingestion code altogether.

Let me know if I'm making sense.

Reply:
Totally makes sense, thanks Lucas! I'll fix this.